### PR TITLE
Adapt adjacent_find to C++20

### PR DESCRIPTION
**Pull request by msimberg**

### CODE REVIEW

File: libs/parallelism/algorithms/include/hpx/parallel/util/invoke_projected.hpp [hunk 1]
```diff
@@ -36,7 +37,47 @@ namespace hpx { namespace parallel { namespace util {
                 pred_, hpx::util::invoke(proj_, std::forward<T>(t)));
         }
 
+        template <typename T>
+        auto operator()(T&& t,
+            T&& u) -> decltype(hpx::util::invoke(std::declval<pred_type>(),
+            hpx::util::invoke(std::declval<proj_type>(), std::forward<T>(t)),
+            hpx::util::invoke(std::declval<proj_type>(), std::forward<T>(u))))
```
Comment (hkaiser) [inline libs/parallelism/algorithms/include/hpx/parallel/util/invoke_projected.hpp:44]: Wouldn't a simple `decltype(auto)` be sufficient? Or do you need the SFINAE induced by the trailing return type calculation?

Comment (msimberg) [reply libs/parallelism/algorithms/include/hpx/parallel/util/invoke_projected.hpp:44]: Good catch. No, I don't think the SFINAE is needed.

**Comment by msimberg** [comment]
This should now finally be ready to go.

**Review by hkaiser — DISMISSED** [review]
Thanks for working on this! 
LGTM, thanks!

**Pull request merged**